docs/best-practices/index.md

```diff
@@ -0,0 +1,438 @@
+
+### Use `--auto-approve` with Caution
+
+Use of the `--auto-approve` feature in Terraform can lead to unintended and potentially destructive changes in your infrastructure. When running Terraform commands, such as `terraform apply`, without the `--auto-approve` flag, Terraform will provide a plan of the changes it intends to make and ask for confirmation before applying those changes.
+
+By using `--auto-approve`, the process of reviewing planned changes to configuration and infrastructure is skipped, and Terraform immediately applies the changes. This can be risky for several reasons:
+
+* **Accidental Changes**: Without reviewing the plan, unintended changes may inadvertently be applied to the environment.  This is especially dangerous in production environments where mistakes can have significant consequences, such as causing breaking changes causing outage or use case failure.
+* **Destructive Changes**: Terraform may plan to destroy resources as part of the update.  Without manual confirmation, unintentional removal of critical configuration or infrastructure may occur.  This applies to both `terraform apply` and `terraform destroy` commands.
+* **Security Implications**: Auto-approving changes without verification increases the risk of security vulnerabilities. For example, sensitive data may unintentionally be exposed, or access policies may be negated or weakened.
+
+To minimize the risks associated with `--auto-approve`, Ping recommends to review the Terraform plan before applying changes.  This ensures that admins have a clear understanding of what modifications Terraform intends to make to live service configuration and infrastructure.
+
+### Store State Securely
+
+When operating production infrastructure with Terraform, the secure storage of Terraform state files is of paramount importance. These files serve as the foundational blueprint of your infrastructure, detailing configurations, credentials, and the current state of resources. Given their critical role, these files inherently contain sensitive information that, if exposed, could be used to gain unauthorized access to user data and manipulation of deployed infrastructure.
+
+To safeguard against these threats, it is vital that robust security measures are implemented around state file storage. This includes:
+- Encrypting the state files to protect their contents during transit and at rest
+- Employing stringent access controls to ensure only authorized personnel can retrieve or alter the state.  If cloud blob storage is used (such as AWS S3), ensure public access is disabled.
+- Leveraging secure storage solutions that offer features like versioning and backups
+
+Hashicorp themselves offer Terraform Cloud that provides secure storage of state out of the box.
```

Comment:
**One option is to use Terraform Cloud from Hashicorp, which provides secure state storage as part of the service.**